packages/core/src/components/tabs2/tabs2.tsx

```diff
@@ -153,6 +153,12 @@ export class Tabs2 extends AbstractComponent<ITabs2Props, ITabs2State> {
     public componentDidUpdate(_prevProps: ITabs2Props, prevState: ITabs2State) {
         if (this.state.selectedTabId !== prevState.selectedTabId) {
             this.moveSelectionIndicator();
+        } else if (prevState.selectedTabId !== undefined && prevState.selectedTabId !== null) {
+            const selectedTabIdIndex = this.props.children[1].map(child => child.props.id).indexOf(this.state.selectedTabId);
+            const prevSelectedTabIdIndex = _prevProps.children[1].map(child => child.props.id).indexOf(prevState.selectedTabId);
```

Comment:
yeah, I don't like that ugly indexing either. I have changed `this.props.children[1]` but I'm not sure how to change `_prevProps.children[1]`